lib/scholar/cluster/k_means.ex

```diff
@@ -8,31 +8,70 @@ defmodule Scholar.Cluster.KMeans do
   @derive {Nx.Container, containers: [:num_iterations, :clusters, :inertia, :labels]}
   defstruct [:num_iterations, :clusters, :inertia, :labels]
 
+  @initialization_schema NimbleOptions.new!(
+                           k_means_plus_plus: [
+                             doc:
+                               "selects initial cluster centroids using sampling based on an empirical probability distribution of the points’ contribution to the overall inertia. This technique speeds up convergence, and is theoretically proven to be O(log(k))-optimal.
+                             "
+                           ],
+                           random: [
+                             doc: """
+                             choose `:num_clusters` observations (rows) at random from data for the initial centroids.
+                             """
+                           ]
+                         )
+
+  @opts_schema NimbleOptions.new!(
+                 num_clusters: [
+                   required: true,
+                   type: :pos_integer,
+                   doc:
+                     "The number of clusters to form as well as the number of centroids to generate."
+                 ],
+                 max_iterations: [
+                   type: :pos_integer,
+                   default: 300,
+                   doc: "Maximum number of iterations of the k-means algorithm for a single run."
+                 ],
+                 num_runs: [
+                   type: :pos_integer,
+                   default: 10,
+                   doc: """
+                    Number of time the k-means algorithm will be run with different centroid seeds.
+                     The final results will be the best output of num_runs runs in terms of inertia.
+                   """
+                 ],
+                 tol: [
+                   type: :float,
+                   default: 1.0e-4,
+                   doc: """
+                    Relative tolerance with regards to Frobenius norm of the difference in
+                    the cluster centers of two consecutive iterations to declare convergence.
+                   """
+                 ],
+                 weights: [
+                   default: nil,
+                   type: {:or, [:atom, {:list, {:or, [:float, :non_neg_integer]}}]},
+                   doc: """
+                   The weights for each observation in x. If equals to `nil`,
+                   all observations are assigned equal weight.
+                   """
+                 ],
+                 init: [
+                   type: {:in, [:k_means_plus_plus, :random]},
+                   default: :k_means_plus_plus,
+                   doc: """
+                   Method for centroid initialization, either of:
+
+                   #{NimbleOptions.docs(@initialization_schema, nest_level: 1)}
```

Comment:
Instead of cascading NimbleOptions, you should just inline those docs here:

```suggestion
                   * `:k_means_plus_plus` - does this
                   * `:random` - does that
```
```

Reply:
When I try this one, the `:k_means_plus_plus` and `:random` are indented wrongly.

Reply:
I will fix it in nimble options.  :)